packages/node-runtime-worker-thread/src/index.ts

```diff
@@ -57,16 +87,49 @@ class WorkerRuntime implements Runtime {
   private async initWorker() {
     const { uri, driverOptions, cliOptions, spawnOptions } = this.initOptions;
 
-    const childProcessProxySrcPath = path.resolve(
-      __dirname,
-      'child-process-proxy.js'
+    this.childProcess = spawn(
+      process.execPath,
+      [this.childProcessProxySrcPath],
+      {
+        stdio: ['inherit', 'inherit', 'pipe', 'ipc'],
+        ...spawnOptions
+      }
     );
 
-    this.childProcess = spawn(process.execPath, [childProcessProxySrcPath], {
-      stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
-      ...spawnOptions
+    const waitForReadyMessage = async() => {
+      let msg: string;
+      while (([msg] = await once(this.childProcess, 'message'))) {
+        if (msg === 'ready') return;
+      }
+    };
+
+    let spawnError = '';
+
+    // eslint-disable-next-line chai-friendly/no-unused-expressions
+    this.childProcess?.stderr?.on('data', (chunk) => {
```

Comment:
```suggestion
    this.childProcess?.stderr?.setEncoding('utf8')?.on('data', (chunk) => {
```

(otherwise you’d see �� when adjacent `chunk` Buffers happen to contain split parts of a UTF-8 characters – not a common edge case, but easy to work around :))

Reply:
Ah, good to know, TIL! Thanks 😊